ui/components/component-library/modal-header/modal-header.tsx

```diff
@@ -54,6 +54,7 @@ export const ModalHeader: React.FC<ModalHeaderProps> = ({
         endAccessory ||
         (onClose && (
           <ButtonIcon
+            data-testid="modal-header-close-button"
```

Comment:
Instead of hardcoding the `data-testid` directly in the atomic component, we can use the `closeButtonProps` to pass it in where needed. This approach ensures `data-testid` values remain unique, helping to avoid false positives and issues from multiple elements sharing the same identifier, which should generally be unique. It will also greatly reduce the amount of file changes in this PR

Reply:
Sure, that works for me. Let me make that change